include/opencv2/opencv.hpp

```diff
@@ -102,4 +101,39 @@
 #include "opencv2/viz.hpp"
 #endif
 
+// Finally CUDA specific entries are checked and added
+#ifdef HAVE_OPENCV_CUDAARITHM
+#include "opencv2/cudaarithm.hpp"
+#endif
+#ifdef HAVE_OPENCV_CUDABGSEGM
+#include "opencv2/cudabgsegm.hpp"
+#endif
+#ifdef HAVE_OPENCV_CUDACODEC
+#include "opencv2/cudacodec.hpp"
+#endif
+#ifdef HAVE_OPENCV_CUDAFEATURES2D
+#include "opencv2/cudafeatures2d.hpp"
+#endif
+#ifdef HAVE_OPENCV_CUDAFILTERS
+#include "opencv2/cudafilters.hpp"
+#endif
+#ifdef HAVE_OPENCV_CUDAIMGPROC
+#include "opencv2/cudaimgproc.hpp"
+#endif
+#ifdef HAVE_OPENCV_CUDALEGACY
+#include "opencv2/cudalegacy.hpp"
```

Comment:
I would exclude the `cudalegacy` module from that list. It hardly depends on CUDA (contains `cuda_runtime.h` include directive), so any code that includes `cudalegacy` must also provide include path for CUDA to compiler. Plus this module is legacy, so it shouldn't be commonly used.